### PR TITLE
Missing contrib test

### PR DESCRIPTION
**Pull request by gpotter2**
This PR will contain some contrib tests + fixes for many un-tested contrib modules.
Also fixes the fact that on Python 2, IPv6 addresses different from ::1 are dissected as unicode, which i randomly triggered

Notes:
- I left a few ones behind. might go over them at some point later
- SEND modules tests are available in a different PR: https://github.com/secdev/scapy/pull/1088

### CODE REVIEW

File: scapy/contrib/ripng.py [hunk 1]
```diff
@@ -24,30 +24,19 @@
...
-            ]
+                    ShortField("null", 0) ]
 
 class RIPngEntry(Packet):
     name = "RIPng entry"
-    fields_desc = [
-                    ConditionalField(IP6Field("prefix", "::"),
-                                            lambda pkt: pkt.metric != 255),
-                    ConditionalField(IP6Field("nexthop", "::"),
-                                            lambda pkt: pkt.metric == 255),
```
Comment (gpotter2) [inline scapy/contrib/ripng.py]: This did not work as metric is below those fields.

File: scapy/contrib/avs.uts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+% Regression tests for the avs module
+
++ Basic AVS test
+
+= Default build, storage and dissection
+
+pkt = AVSWLANHeader()/Dot11()/Dot11Auth()
+wrpcap("test.pcap", pkt)
```
Comment (guedou) [inline scapy/contrib/avs.uts]: Please use `get_temp_file()`, here and in other parts of your PR.

File: scapy/contrib/carp.py [hunk 3]
```diff
@@ -20,64 +20,55 @@
...
+                    XIntField("hmac1", 0),
+                    XIntField("hmac2", 0),
+                    XIntField("hmac3", 0),
+                    XIntField("hmac4", 0),
+                    XIntField("hmac5", 0) ]
 
     def post_build(self, pkt, pay):
         if self.chksum == None:
             pkt = pkt[:6] + struct.pack("!H", checksum(pkt)) + pkt[8:]
-
```
[the remark this answers is not in the document]
Comment (gpotter2) [reply scapy/contrib/carp.py]: Right !

File: scapy/contrib/carp.py [hunk 4]
```diff
@@ -20,64 +20,55 @@
...
-        XShortField("chksum", 0),
-        XIntField("counter1", 0),
-        XIntField("counter2", 0),
-        XIntField("hmac1", 0),
-        XIntField("hmac2", 0),
-        XIntField("hmac3", 0),
-        XIntField("hmac4", 0),
-        XIntField("hmac5", 0)
-    ]
+                    BitField("type", 4, 4),
```
[the remark this answers is not in the document]
Comment (gpotter2) [reply scapy/contrib/carp.py]: Sorry for that

File: scapy/contrib/avs.py [hunk 5]
```diff
@@ -21,6 +21,8 @@
 from scapy.fields import *
 from scapy.layers.dot11 import *
 
+LINKTYPE_IEEE802_11_AVS = 163
```
Comment (p-l-) [inline scapy/contrib/avs.py]: Please create a constant value `DLT_IEEE802_11_RADIO_AVS = 163` in `data.py` and use it instead.

File: scapy/contrib/etherip.uts [hunk 6]
```diff
@@ -0,0 +1,7 @@
++ EtherIP Contrib tests
+
+= Basic EtherIP test
+
+pkt = Ether(b'\x99\xc1o\xd2\xf5c\x9d\xb7\xd0\xc2\xe0\xd3\x08\x00E\x00\x00@\x00\x01\x00\x00@a,\xf3B\x83\x17\xc6\xad\xc2E^0\x00\xd5/\xf26\xab\xe2\x9f\xb4tD\xa4\x98\x08\x00E\x00\x00\x1c\x00\x01\x00\x00@\x01W-\xe7\x98H\xfa\xad\xc2E^\x08\x00\xf7\xff\x00\x00\x00\x00')
+assert ICMP in pkt
+assert EtherIP in pkt
```
Comment (p-l-) [inline scapy/contrib/etherip.uts]: Missing newline at EOF.

File: scapy/contrib/rsvp.uts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+pkt = Ether()/IP()/RSVP()/RSVP_Object()/RSVP_SessionAttrb(Name="test")
+pkt = Ether(raw(pkt))
+assert RSVP_SessionAttrb in pkt
+assert pkt.Name == b"test"
+
+= Master dissection
+
+pkt = Ether(b"\x00\x90\x92\x9d\x94\x01\x00\xd0c\xc3\xb8G\x08\x00E\x00\x00\x80\x8ad\x00\x00\xff.\x8c\xe7\xd2\x00\x00\x02\xd2\x00\x00\x01\x10\x02\xeb\xfa\xff\x00\x00l\x00\x10\x01\x07\x10\x02\x02\x02\x00\x00\x00\x01\x11\x03\x03\x03\x00\x0c\x03\x01\xd2\x00\x00\x02\x00\x00\x00\x00\x00\x08\x05\x01\x00\x00u0\x00\x08\x08\x01\x00\x00\x00\x12\x00$\t\x02\x00\x00\x00\x07\x05\x00\x00\x06\x7f\x00\x00\x05I\x18\x96\x80Dz\x00\x00\x7f\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x0c\n\x07\x11\x03\x03\x03\x00\x00'\x11\x00\x08\x10\x01\x00\x00\x00\x10\x03\x06-\xad")
+assert RSVP_Time in pkt
+assert pkt[RSVP_Time].refresh == 30000
```
Comment (p-l-) [inline scapy/contrib/rsvp.uts]: Missing newline at EOF.

File: scapy/contrib/spbm.uts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+backboneServiceID = SPBM(prio=1,isid=20011)
+customerEther     = Ether(dst='00:1b:4f:5e:ca:00',src='00:00:00:00:00:01',type=0x8100)
+customerDot1Q     = Dot1Q(prio=1,vlan=11,type=0x0800)
+customerIP        = IP(src='10.100.11.10',dst='10.100.12.10',id=0x0629,len=106)
+customerUDP       = UDP(sport=1024,dport=1025,chksum=0,len=86)
+
+pkt = backboneEther/backboneDot1Q/backboneServiceID/customerEther/customerDot1Q/customerIP/customerUDP/"Payload"
+pkt = Ether(raw(pkt))
+assert SPBM in pkt
+assert pkt[SPBM].payload.payload.payload.src == '10.100.11.10'
```
Comment (p-l-) [inline scapy/contrib/spbm.uts]: Missing newline at EOF.

Comment (gpotter2) [reply scapy/contrib/spbm.uts]: Added

File: scapy/contrib/avs.uts [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+pkt1 = rdpcap(_filepath)[0]
+assert raw(pkt) == raw(pkt1)
+assert AVSWLANHeader in pkt
+assert Dot11 in pkt
+assert Dot11Auth in pkt
+
+try:
+    os.remove(_filepath)
+except:
+    pass
```
Comment (p-l-) [inline scapy/contrib/avs.uts]: Sorry I missed this one. BTW you don't need an extra blank line at the end of file, you just want that the last line ends with a '\n' (and sorry to bother you with this...).

File: scapy/contrib/chdlc.uts [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+pkt = CHDLC()/SLARP(type=0, address="192.168.0.131", mask="255.255.0.0")
+pkt = CHDLC(raw(pkt))
+assert pkt[SLARP].address == "192.168.0.131"
+
+= Build keepalive
+
+pkt = CHDLC()/SLARP(type=2, mysequence=123, yoursequence=123456789, reliability=555)
+pkt = CHDLC(raw(pkt))
+assert pkt[SLARP].yoursequence == 123456789
```
Comment (p-l-) [inline scapy/contrib/chdlc.uts]: Also, this one...

Comment (gpotter2) [reply scapy/contrib/chdlc.uts]: Updated !

File: scapy/contrib/avs.uts (hunk 9 above)
Comment (gpotter2) [reply scapy/contrib/avs.uts]: Done

File: scapy/contrib/rsvp.uts (hunk 7 above)
Comment (gpotter2) [reply scapy/contrib/rsvp.uts]: Added

File: scapy/contrib/etherip.uts (hunk 6 above)
Comment (gpotter2) [reply scapy/contrib/etherip.uts]: Added

**Comment by p-l-** [comment]
@guedou OK for me.

File: scapy/contrib/rsvp.py [hunk 11]
```diff
@@ -149,25 +150,29 @@ def guess_payload_class(self, payload):
 
 class RSVP_Data(Packet):
     name = "Data"
+    overload_fields = { RSVP_Object: { "Class": 0x01}}
```
Comment (guedou) [inline scapy/contrib/rsvp.py]: Your `overload_fields` in this file use a strange spacing.

File: scapy/contrib/ldp.uts [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+assert LDPHello in pkt4
+assert LDPAddressWM in pkt5
+assert LDPLabelMM in pkt6
+
+= Basic dissection
+
+pkt = Ether(b'AJH\x18\x07\xfa\xd0P\x99V\xdd\xf9\x08\x00E\x00\x006\x00\x01\x00\x00@\x11:\x96(\x9d\r\xd3\xc1\x1eq\x10\x02\x86\x02\x86\x00"5\xa1\x00\x01\x00\x16\x7f\x00\x00\x01\x00\x00\x01\x00\x00\x0c\x00\x00\x00\x00\x04\x00\x00\x04\x00\xb4\x00\x00')
+assert LDPHello in pkt
+assert pkt[LDP].id == "127.0.0.1"
+assert pkt[LDPHello].params == [180, 0, 0]
```
Comment (guedou) [inline scapy/contrib/ldp.uts]: Missing newline at EOF.

**Comment by guedou** [comment]
@p-l- I made two small comments.

**Comment by gpotter2** [comment]
Updated !
Had to rebase to fix coverage. Only the last commit is new

**Comment by gpotter2** [comment]
@guedou Hi, could you restart the failing test, please ?
This PR should be ready to merge :)

**Comment by guedou** [comment]
Done & approved.

**Pull request merged**